File: backend/app/schemas/mindmap.py
"""思维导图相关响应模型（Phase 6）。"""
from pydantic import BaseModel, ConfigDict


class MindNode(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    name: str
    kind: str = "topic"
    entity_id: int | None = None
    meta: dict = {}
    children: list["MindNode"] = []


class MindmapResponse(BaseModel):
    mode: str  # llm / local
    root: MindNode
    node_count: int

File: frontend/src/api/mindmap.ts
import request from '@/utils/request'

export interface MindNode {
  name: string
  kind: string
  entity_id: number | null
  meta: Record<string, any>
  children: MindNode[]
}

export interface MindmapData {
  mode: string
  root: MindNode
  node_count: number
}

/** 获取当前用户的思维导图（由知识图谱投影生成）。category 可选，按分类生成。 */
export function getMindmap(category?: string): Promise<MindmapData> {
  return request.get('/mindmap', { params: { category } })
}
